src/HieDb/Types.hs

```diff
@@ -34,7 +34,7 @@ import Database.SQLite.Simple.FromField
 
 import qualified Text.ParserCombinators.ReadP as R
 
-data HieDb = HieDb { getConn :: Connection, getDbDynFlags :: Maybe DynFlags }
```

Comment:
I don't see much utility in having `Maybe DynFlags` as part of HieDb.
There were 3 incomplete pattern matches like `Just dynFlags = getDbDynFlags hieDb`
Instead I suggest including DynFlags in `withHieDbAndFlags` above

Reply:
Before [this commit](https://github.com/wz1000/HieDb/commit/79aaaae3bd8e53cca94cb3a1e71f020969e4015f) we were using the dynflags to pretty print types while indexing.

We might want to pretty print stuff like this in the future, so I'm not very sure about this change.

However, it could be the case that keeping a copy of the dynflags in the db state is not a good idea, and in the future we should just rely on the caller to supply the DynFlags when it is needed. This does indeed seem like a good idea from the ghcide perspective.

Reply:
In my mind it's better to be upfront about what the function needs.
If it needs `DynFlags`, it should have it as explicit parameter.
If you hold `HieDb` value in hand you can't be sure if it contains `Just DynaFlags` or `Nothing`.
This way you have to solve the problem "what do I do if I don't have DynFLags".
Think about it and let me know if you don't think this is a good idea, we can close this PR.